### PR TITLE
Make member vars between Density and ADDensity consistent

### LINKED ISSUE #17229: Make Density and ADDensity consistent
## Reason
The member vars between Density and ADDensity should be consistent to allow for combination in the future. This came about by #17227.

## Design
Make the member vars of Density and ADDensity consistent.

## Impact
Consistent API that allows for combining into a templated object in the future.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #17229

@dschwen ask and thou shall receive

### CODE REVIEW

**Review by dschwen — DISMISSED** [review]

File: modules/misc/src/materials/Density.C [hunk 1]
```diff
@@ -32,42 +32,31 @@ Density::validParams()
...
   : Material(parameters),
-    _is_coupled(true),
-    _disp_r(isCoupled("displacements") ? coupledValue("displacements", 0) : _zero),
+    _is_coupled(isCoupled("displacements")),
+    _coord_system(getBlockCoordSystem()),
+    _disp_r(_is_coupled ? coupledValue("displacements", 0) : _zero),
     _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
-    _orig_density(getParam<Real>("density")),
+    _initial_density(getParam<Real>("density")),
+    _grad_disp(_is_coupled ? coupledGradients("") : std::vector<const VariableGradient *>()),
```
Comment (dschwen) [inline modules/misc/src/materials/Density.C:40]:
```suggestion
    _grad_disp(_is_coupled ? coupledGradients("displacements") : std::vector<const VariableGradient *>()),
```

**Review by dschwen — CHANGES_REQUESTED** [review]

File: modules/misc/src/materials/Density.C (hunk 1 above)
Comment (loganharbour) [reply modules/misc/src/materials/Density.C:40]: oops I'm bad at this

**Comment by loganharbour** [comment]
@dschwen where'd ya go

**Review by dschwen — APPROVED** [review]
:+1:

**Pull request merged**